### PR TITLE
nixos/{polkit,systemd-coredump}: guard static gid behind state version

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
Suppresses the annoying warning for existing systems.
See https://github.com/NixOS/nixpkgs/pull/215345#issuecomment-1439986349

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/systemd/coredump.nix [hunk 1]
```diff
@@ -67,7 +67,7 @@ in {
         group = "systemd-coredump";
       };
       users.groups.systemd-coredump = {
-        gid = config.ids.gids.systemd-coredump;
+        gid = if (lib.versionAtLeast config.system.stateVersion "23.05") then config.ids.gids.systemd-coredump else null;
```
Comment (ncfavier) [inline nixos/modules/system/boot/systemd/coredump.nix:70]:
```suggestion
        gid = mkIf (lib.versionAtLeast config.system.stateVersion "23.05") config.ids.gids.systemd-coredump;
```

**Review by ncfavier — APPROVED** [review]

**Comment by winterqt** [comment]
Sorry, wasn't able to get to this in time, but for the future, I'd recommend adding your justification to both the PR and the commit description(s), as (for example) these changes aren't so obvious.

**Comment by NickCao** [comment]
> also note that this does not _fix_ the problem of warnings appearing

It does fix the issue of warning appearing without user intervention. If a user deliberately bump the stateVersion, it means he/she is willing to do any resulting manual migration, and the warning is a reminder for that. 

> ioo neither this nor the preceding PR should've been merged in the absence of strong reasons why a globally reserved gid for these is _required_

I raised the same concern in https://github.com/NixOS/nixpkgs/pull/215345#issuecomment-1439999849.

**Comment by winterqt** [comment]
> I raised the same concern in https://github.com/NixOS/nixpkgs/pull/215345#issuecomment-1439999849.

I mean this in the nicest way possible: why did you merge it, then? Everyone could have discussed it instead of making the "problem" worse.

**Comment by winterqt** [comment]
Sorry, you questioned it after merging.

**Comment by NickCao** [comment]
Unless we have a final resolution to fully deprecate static uid/gid for systemd-* users, I think having the consistency/aesthetic is beneficial, not worsening. We should not leave these minor frustrations floating forever, and guarding them behind a stateVersion, is what I see as the most appropriate action now.

**Pull request merged**